### ISSUE TITLE
Fails to build development environment

### ISSUE THREAD

**Issue by d4hines** [ocurrent/current-bench#88]
Hello,

current-bench fails to start for me.

I'm on Windows 10, running Docker version 20.10.5.

The contents of ./environments/development.env are:
```
OCAML_BENCH_DOCKER_CPU=1
OCAML_BENCH_DB_PASSWORD=docker
OCAML_BENCH_GRAPHQL_KEY=secret
OCAML_BENCH_TARGET_ARCH=amd64
```

I run `make start-development` and get the following output (errors way down at bottom):
```
PS C:\Users\d4hin\repos\current-bench> make start-development
docker-compose \
        --project-name="current-bench" \
        --file=./environments/development.docker-compose.yaml \
        --env-file=./development.env \
        up \
        --remove-orphans \
        --build
Building db
failed to get console mode for stdout: The handle is invalid.
[+] Building 11.2s (7/7) FINISHED
 => [internal] load build definition from Dockerfile                       0.0s 
 => => transferring dockerfile: 151B                                       0.0s 
 => [internal] load .dockerignore                                          0.0s 
 => => transferring context: 2B                                            0.0s 
 => [internal] load metadata for docker.io/library/postgres:12             1.7s 
 => [internal] load build context                                          0.0s 
 => => transferring context: 87B                                           0.0s 
 => [1/2] FROM docker.io/library/postgres:12@sha256:2f3f78532c9cc5435d1cf  9.2s 
 => => resolve docker.io/library/postgres:12@sha256:2f3f78532c9cc5435d1cf  0.0s 
 => => sha256:2f3f78532c9cc5435d1cf9c8f5fb1409f9abd43e5b7 1.86kB / 1.86kB  0.0s
 => => sha256:75646c2fb4101d306585c9b106be1dfa7d82720ba 27.14MB / 27.14MB  1.0s
 => => sha256:7e98825f6d67159fc035c0db985619c0635cc7edb89 1.80kB / 1.80kB  0.3s
 => => sha256:6ef141d922bc14123d24ddec34aa11d0f360702a2 10.63kB / 10.63kB  0.0s
 => => sha256:4217f50ff2ca28b7c3dddb25372379c97093df532c7 3.25kB / 3.25kB  0.0s
 => => sha256:2355d0ffeb5531909b8910928bcbaa874e9464657c7 4.18MB / 4.18MB  0.4s
 => => sha256:cfd3ce06be45a467d795e78a52a17330a56a7b85a3b 1.42MB / 1.42MB  0.5s
 => => sha256:c7b7bb83e8f78baa3e4e402b1704b6df53db00bec8b 7.97MB / 7.97MB  0.9s
 => => sha256:c6786930510827ff4067b1103131548886adcd7 391.17kB / 391.17kB  0.6s
 => => sha256:19614baa7ddd150dc7115b541a85cec0e8178fc9c6e62c8 149B / 149B  0.6s
 => => sha256:af508737d8137d59629c343294044cb7f854220431d 3.05kB / 3.05kB  0.7s
 => => sha256:562171e5e21c288ecc3ae718a64e16c2f17902deb 73.34MB / 73.34MB  6.1s
 => => sha256:9a1da7929f851b2f7db64c04a23cf5b2c7377b0db64 9.02kB / 9.02kB  1.0s
 => => extracting sha256:75646c2fb4101d306585c9b106be1dfa7d82720baabe1c75  1.3s
 => => sha256:83302896f1018b739b0cdcdd817cbc82c823cf8651cffd6 200B / 200B  1.2s
 => => sha256:1f2e5b2f6e536af3111e90491c622a3bd93179b70459f09 129B / 129B  1.2s
 => => sha256:8892c210e4afc8a8f8ece7cd0bcccfee3b234dd09b6 4.40kB / 4.40kB  1.4s
 => => sha256:46ddd2f5c65450c2987d3d2526b0a0b7d6f1655d312fb31 120B / 120B  1.4s
 => => extracting sha256:2355d0ffeb5531909b8910928bcbaa874e9464657c71213a  0.2s
 => => extracting sha256:7e98825f6d67159fc035c0db985619c0635cc7edb89aee89  0.0s
 => => extracting sha256:cfd3ce06be45a467d795e78a52a17330a56a7b85a3b3a25b  0.1s
 => => extracting sha256:c7b7bb83e8f78baa3e4e402b1704b6df53db00bec8b57266  0.3s
 => => extracting sha256:c6786930510827ff4067b1103131548886adcd7a25a012ba  0.1s
 => => extracting sha256:19614baa7ddd150dc7115b541a85cec0e8178fc9c6e62c82  0.0s
 => => extracting sha256:af508737d8137d59629c343294044cb7f854220431d23fc4  0.0s
 => => extracting sha256:562171e5e21c288ecc3ae718a64e16c2f17902deb1011cad  2.5s
 => => extracting sha256:9a1da7929f851b2f7db64c04a23cf5b2c7377b0db64b3b67  0.0s
 => => extracting sha256:1f2e5b2f6e536af3111e90491c622a3bd93179b70459f098  0.0s
 => => extracting sha256:83302896f1018b739b0cdcdd817cbc82c823cf8651cffd68  0.0s
 => => extracting sha256:8892c210e4afc8a8f8ece7cd0bcccfee3b234dd09b6b2637  0.0s
 => => extracting sha256:46ddd2f5c65450c2987d3d2526b0a0b7d6f1655d312fb310  0.0s
 => [2/2] COPY ./init.sql /docker-entrypoint-initdb.d/init.sql             0.1s
 => exporting to image                                                     0.0s
 => => exporting layers                                                    0.0s
 => => writing image sha256:65222942325e8b01767e984270bbcbc7150b551d35577  0.0s
 => => naming to docker.io/library/current-bench_db                        0.0s
Successfully built 65222942325e8b01767e984270bbcbc7150b551d35577e19aa4f3177c8caa3e6
Pulling graphql-engine (hasura/graphql-engine:v1.3.3)...
v1.3.3: Pulling from hasura/graphql-engine
fcf970b8ac66: Pull complete
Digest: sha256:814e36b355177046195edd795ca561fe762a2ae117129f185a26f9d3f609efb9
Status: Downloaded newer image for hasura/graphql-engine:v1.3.3
Building frontend
failed to get console mode for stdout: The handle is invalid.
[+] Building 42.9s (17/17) FINISHED
 => [internal] load build definition from Dockerfile                       0.0s 
 => => transferring dockerfile: 554B                                       0.0s 
 => [internal] load .dockerignore                                          0.0s 
 => => transferring context: 52B                                           0.0s 
 => [internal] load metadata for docker.io/library/nginx:stable-alpine     0.7s 
 => [internal] load metadata for docker.io/library/node:lts                0.8s 
 => [builder 1/7] FROM docker.io/library/node:lts@sha256:f6b9ff4caca9d4f  18.1s 
 => => resolve docker.io/library/node:lts@sha256:f6b9ff4caca9d4f0a331a882  0.0s 
 => => sha256:abea835c0b3b0ef244d92a1076602ed0891a35b1b77 7.83kB / 7.83kB  0.0s 
 => => sha256:da61ad49fa9961d6dfe53dc067fd690133f205498 11.29MB / 11.29MB  0.7s
 => => sha256:af62e04c0f85ed10801df4314d8cb9ba5e6c391f6a3 4.34MB / 4.34MB  0.5s
 => => sha256:f6b9ff4caca9d4f0a331a882e560df242eb332b7bbbed2f 776B / 776B  0.0s
 => => sha256:61e6ae4231ac78eb737a10e70d8925e92e7a35ec0c3 2.21kB / 2.21kB  0.0s
 => => sha256:00168f89dbe8f3c9985e536784c27517f6cc35ea5 45.38MB / 45.38MB  2.9s
 => => sha256:aae835e2c68ef1d48a4ffef72547a79926019aec6 49.79MB / 49.79MB  3.6s
 => => sha256:5254dc317968c943c4321cdbf0364a6a47c8332 214.35MB / 214.35MB  8.3s
 => => sha256:b3f76fdfa8e8784238b141aaa54fa1862aef5348665 4.19kB / 4.19kB  2.9s
 => => extracting sha256:00168f89dbe8f3c9985e536784c27517f6cc35ea56263469  1.9s
 => => sha256:5681b30a17f656444f1bd37c07ee6f594c7e2e26c 34.58MB / 34.58MB  5.5s
 => => sha256:da9a613e2daeaae78b9cd247e52f3f731b701730d7a 2.38MB / 2.38MB  3.9s
 => => sha256:67891041a0a2d985480f4934fc94bc0b180f9f27741bd5c 292B / 292B  4.1s
 => => extracting sha256:da61ad49fa9961d6dfe53dc067fd690133f205498e8045e8  0.4s
 => => extracting sha256:af62e04c0f85ed10801df4314d8cb9ba5e6c391f6a3c1db1  0.2s
 => => extracting sha256:aae835e2c68ef1d48a4ffef72547a79926019aec6a2bbeb5  2.4s
 => => extracting sha256:5254dc317968c943c4321cdbf0364a6a47c833296b3f45e3  7.6s
 => => extracting sha256:b3f76fdfa8e8784238b141aaa54fa1862aef534866543116  0.0s
 => => extracting sha256:5681b30a17f656444f1bd37c07ee6f594c7e2e26c4ec796e  1.6s
 => => extracting sha256:da9a613e2daeaae78b9cd247e52f3f731b701730d7ab8bf6  0.1s
 => => extracting sha256:67891041a0a2d985480f4934fc94bc0b180f9f27741bd5c2  0.0s
 => [stage-1 1/4] FROM docker.io/library/nginx:stable-alpine@sha256:ec3f6  5.7s
 => => resolve docker.io/library/nginx:stable-alpine@sha256:ec3f6e4587a20  0.0s
 => => sha256:ec3f6e4587a2053f406b1b4e027c260e82e4e32c256 1.65kB / 1.65kB  0.0s
 => => sha256:7409b874fed478250a841bc67aa1451df7163543cf6 1.36kB / 1.36kB  0.0s
 => => sha256:d0d03989cd8e044493922e446203a9893eb662a3279 7.98kB / 7.98kB  0.0s
 => => sha256:9b794450f7b6db7c944ba1f4161edb68cb535052fe7 2.82MB / 2.82MB  4.4s
 => => extracting sha256:9b794450f7b6db7c944ba1f4161edb68cb535052fe7db8ac  0.1s
 => => sha256:71c851db865893ed287bbecc2934e756a25e4e78bab 6.67MB / 6.67MB  4.9s
 => => extracting sha256:71c851db865893ed287bbecc2934e756a25e4e78babcd419  0.3s
 => => sha256:fc9aed49b78e299c67c0ee890e26326969df6c564fda2b2 902B / 902B  5.0s
 => => sha256:bb846ca9b45ab911d714856abaca670ed6a798711e73325 665B / 665B  5.1s
 => => sha256:281095db834a17364db464c05304d89414f2ab118283136 600B / 600B  5.0s
 => => extracting sha256:281095db834a17364db464c05304d89414f2ab118283136b  0.0s
 => => extracting sha256:fc9aed49b78e299c67c0ee890e26326969df6c564fda2b28  0.0s
 => => extracting sha256:bb846ca9b45ab911d714856abaca670ed6a798711e73325d  0.0s
 => [internal] load build context                                          0.0s
 => => transferring context: 332.87kB                                      0.0s
 => [stage-1 2/4] COPY ./nginx.conf /etc/nginx/nginx.conf                  0.2s
 => [stage-1 3/4] RUN rm -rf /usr/share/nginx/html/*                       0.3s
 => [builder 2/7] WORKDIR /app                                             0.4s
 => [builder 3/7] COPY package.json yarn.lock /app/                        0.0s
 => [builder 4/7] RUN yarn install                                        14.7s
 => [builder 5/7] COPY . /app/                                             0.0s
 => [builder 6/7] RUN yarn build                                           2.1s
 => [builder 7/7] RUN yarn bundle                                          6.5s
 => [stage-1 4/4] COPY --from=builder /app/dist /usr/share/nginx/html      0.0s
 => exporting to image                                                     0.1s
 => => exporting layers                                                    0.0s
 => => writing image sha256:c536e74e0967b7344ff7988a302a7ae2068e7bdfcd00f  0.0s
 => => naming to docker.io/library/current-bench_frontend                  0.0s
Successfully built c536e74e0967b7344ff7988a302a7ae2068e7bdfcd00fa74233b8a508ada4d16
Building pipeline
failed to get console mode for stdout: The handle is invalid.
[+] Building 170.7s (22/22) FINISHED
 => [internal] load build definition from Dockerfile                       0.0s 
 => => transferring dockerfile: 1.78kB                                     0.0s 
 => [internal] load .dockerignore                                          0.0s 
 => => transferring context: 2B                                            0.0s 
 => [internal] load metadata for docker.io/library/debian:stable-slim      0.9s 
 => [internal] load metadata for docker.io/ocaml/opam:debian-ocaml-4.12    0.9s 
 => [internal] load build context                                          0.0s 
 => => transferring context: 54.32kB                                       0.0s 
 => [build 1/8] FROM docker.io/ocaml/opam:debian-ocaml-4.12@sha256:e39cf  49.1s 
 => => resolve docker.io/ocaml/opam:debian-ocaml-4.12@sha256:e39cf5ddf42c  0.0s
 => => sha256:004f1eed87df3f75f5e2a1a649fa7edd7f713d130 50.43MB / 50.43MB  2.8s
 => => sha256:b704dfdc404aa80ee420ceb91d1974441bb41b 286.70MB / 286.70MB  20.9s
 => => sha256:dbb52d71bac56d7b6969ac8b0310d33588f700 189.72MB / 189.72MB  14.2s
 => => sha256:e39cf5ddf42ceff246eee0402466a81e47a1c29567a 1.66kB / 1.66kB  0.0s
 => => sha256:7671537e58b4c6a388caee4d445b9a16476fbb3eb6fd1d2 956B / 956B  0.0s
 => => sha256:f7c1669319a4c7bcd2269e33e7bf6950679512069 10.47kB / 10.47kB  0.0s
 => => extracting sha256:004f1eed87df3f75f5e2a1a649fa7edd7f713d1300532fd0  2.2s
 => => extracting sha256:b704dfdc404aa80ee420ceb91d1974441bb41b37fce81fc  20.2s
 => => extracting sha256:dbb52d71bac56d7b6969ac8b0310d33588f7009a3692b933  6.4s
 => [stage-1 1/8] FROM docker.io/library/debian:stable-slim@sha256:f31a1f  6.4s
 => => resolve docker.io/library/debian:stable-slim@sha256:f31a1f9e274ebf  0.0s
 => => sha256:f31a1f9e274ebf9a97b964c9aed3386e1720d56a3d2 1.85kB / 1.85kB  0.0s
 => => sha256:65c240929acadafb6cdf7447d7e3ba1abc214a8b008fe4e 529B / 529B  0.0s
 => => sha256:f052c451335940bea31ab0e58feff9fc657959c4915 1.46kB / 1.46kB  0.0s
 => => sha256:7fa62122c34635d215ecb61e3d515d9eb1676a0f2 27.14MB / 27.14MB  5.0s
 => => extracting sha256:7fa62122c34635d215ecb61e3d515d9eb1676a0f258d352e  1.3s
 => [stage-1 2/8] RUN apt-get update     && apt-get install --no-install  26.2s
 => [stage-1 3/8] WORKDIR /app                                             0.0s
 => [stage-1 4/8] COPY ./aslr_seccomp.json /app/aslr_seccomp.json          0.0s
 => [stage-1 5/8] COPY ./db/migrations /app/db/migrations                  0.0s
 => [build 2/8] RUN sudo apt-get update &&     sudo apt-get install -qq   16.0s
 => [build 3/8] WORKDIR /mnt/project                                       0.0s
 => [build 4/8] COPY --chown=opam:opam pipeline.opam pipeline.opam         0.0s
 => [build 5/8] RUN opam install -y --deps-only -t .                     100.1s
 => [build 6/8] COPY --chown=opam . .                                      0.0s
 => [build 7/8] RUN sudo chown opam .                                      0.5s
 => [build 8/8] RUN opam exec -- dune build --profile=release bin/main.ex  2.2s
 => [stage-1 6/8] COPY --from=build /home/opam/.opam/4.12/bin/omigrate /a  0.1s
 => [stage-1 7/8] COPY ./scripts/wait-for /app/bin/wait-for                0.0s
 => [stage-1 8/8] COPY --from=build /mnt/project/_build/default/bin/main.  0.1s
 => exporting to image                                                     1.5s
 => => exporting layers                                                    1.5s
 => => writing image sha256:943cfc7ba022701594600f0d26fd96ada9ff5fbbac566  0.0s
 => => naming to docker.io/library/current-bench_pipeline                  0.0s
Successfully built 943cfc7ba022701594600f0d26fd96ada9ff5fbbac566a9c7ad3c80558708e94
Creating current-bench_db_1 ... done
Creating current-bench_pipeline_1       ... done
Creating current-bench_db-migrate_1     ... done
Creating current-bench_graphql-engine_1 ... done
Creating current-bench_frontend_1       ... done
Attaching to current-bench_db_1, current-bench_graphql-engine_1, current-bench_db-migrate_1, current-bench_pipeline_1, current-bench_frontend_1
db-migrate_1      | standard_init_linux.go:219: exec user process caused: no such file or directory
db_1              | The files belonging to this database system will be owned by user "postgres".
db_1              | This user must also own the server process.
db_1              |
db_1              | The database cluster will be initialized with locale "en_US.utf8".
db_1              | The default database encoding has accordingly been set to "UTF8".
db_1              | The default text search configuration will be set to "english".
db_1              |
db_1              | Data page checksums are disabled.
db_1              |
db_1              | fixing permissions on existing directory /var/lib/postgresql/data ... ok
db_1              | creating subdirectories ... ok
db_1              | selecting dynamic shared memory implementation ... posix
db_1              | selecting default max_connections ... 100
db_1              | selecting default shared_buffers ... 128MB
db_1              | selecting default time zone ... Etc/UTC
db_1              | creating configuration files ... ok
db_1              | running bootstrap script ... ok
db_1              | performing post-bootstrap initialization ... ok
db_1              | syncing data to disk ... ok
db_1              |
db_1              |
db_1              | Success. You can now start the database server using:
db_1              |
db_1              |     pg_ctl -D /var/lib/postgresql/data -l logfile start
db_1              |
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:12:48.629+0000","level":"info","detail":{"kind":"server_configuration","info":{"live_query_options":{"batch_size":100,"refetch_delay":1},"transaction_isolation":"ISOLATION LEVEL READ COMMITTED","plan_cache_options":{"plan_cache_size":4000},"enabled_log_types":["http-log","websocket-log","startup","webhook-log","query-log"],"server_host":"HostAny","enable_allowlist":false,"log_level":"info","auth_hook_mode":null,"use_prepared_statements":true,"unauth_role":null,"stringify_numeric_types":false,"enabled_apis":["metadata","graphql","config","pgdump"],"enable_telemetry":true,"enable_console":true,"auth_hook":null,"jwt_secret":null,"cors_config":{"allowed_origins":"*","disabled":false,"ws_read_cookie":null},"console_assets_dir":null,"admin_secret_set":true,"port":8080}}}
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:12:48.629+0000","level":"info","detail":{"kind":"postgres_connection","info":{"retries":1,"database_url":"postgres://docker:...@db:5432/docker"}}}
graphql-engine_1  | {"type":"pg-client","timestamp":"2021-04-02T20:12:48.629+0000","level":"warn","detail":{"message":"postgres 
connection failed, retrying(0)."}}
graphql-engine_1  | {"type":"pg-client","timestamp":"2021-04-02T20:12:48.629+0000","level":"warn","detail":{"message":"postgres 
connection failed, retrying(1)."}}
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:12:48.629+0000","level":"error","detail":{"kind":"catalog_migrate","info":{"internal":"could not connect to server: Connection refused\n\tIs the server running on host \"db\" (172.18.0.2) and accepting\n\tTCP/IP connections on port 5432?\n","path":"$","error":"connection error","code":"postgres-error"}}}
graphql-engine_1  | {"internal":"could not connect to server: Connection refused\n\tIs the server running on host \"db\" (172.18.0.2) and accepting\n\tTCP/IP connections on port 5432?\n","path":"$","error":"connection error","code":"postgres-error"}       
db_1              | initdb: warning: enabling "trust" authentication for local connections
db_1              | You can change this by editing pg_hba.conf or using the option -A, or
db_1              | --auth-local and --auth-host, the next time you run initdb.
pipeline_1        |    application [INFO] Logging ready.
db_1              | waiting for server to start....2021-04-02 20:12:48.618 UTC [47] LOG:  starting PostgreSQL 12.6 (Debian 12.6-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
pipeline_1        |    application [ERROR] Database connection error:
pipeline_1        | Connection failure: could not connect to server: Connection refused
pipeline_1        |     Is the server running on host "db" (172.18.0.2) and accepting
pipeline_1        |     TCP/IP connections on port 5432?
pipeline_1        |
db_1              | 2021-04-02 20:12:48.620 UTC [47] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1              | 2021-04-02 20:12:48.634 UTC [48] LOG:  database system was shut down at 2021-04-02 20:12:48 UTC
pipeline_1        | pipeline: internal error, uncaught exception:
pipeline_1        |           Postgresql.Error(_)
pipeline_1        |           Raised at Pipeline__Db_util.check_connection in file "lib/db_util.ml", line 10, characters 6-15   
pipeline_1        |           Called from Pipeline.v in file "lib/pipeline.ml", line 245, characters 2-36
pipeline_1        |           Called from Cmdliner_term.app.(fun) in file "cmdliner_term.ml", line 25, characters 19-24
pipeline_1        |           Called from Cmdliner.Term.run in file "cmdliner.ml", line 117, characters 32-39
db_1              | 2021-04-02 20:12:48.637 UTC [47] LOG:  database system is ready to accept connections
db_1              |  done
db_1              | server started
db_1              | CREATE DATABASE
db_1              |
db_1              |
db_1              | /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/init.sql
db_1              | GRANT
db_1              |
db_1              |
current-bench_db-migrate_1 exited with code 1
db_1              | 2021-04-02 20:12:48.919 UTC [47] LOG:  received fast shutdown request
db_1              | waiting for server to shut down....2021-04-02 20:12:48.921 UTC [47] LOG:  aborting any active transactions  
db_1              | 2021-04-02 20:12:48.922 UTC [47] LOG:  background worker "logical replication launcher" (PID 54) exited with exit code 1
db_1              | 2021-04-02 20:12:48.922 UTC [49] LOG:  shutting down
db_1              | 2021-04-02 20:12:48.935 UTC [47] LOG:  database system is shut down
db_1              |  done
db_1              | server stopped
db_1              |
db_1              | PostgreSQL init process complete; ready for start up.
db_1              |
db_1              | 2021-04-02 20:12:49.027 UTC [1] LOG:  starting PostgreSQL 12.6 (Debian 12.6-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bitgnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
db_1              | 2021-04-02 20:12:49.028 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
db_1              | 2021-04-02 20:12:49.028 UTC [1] LOG:  listening on IPv6 address "::", port 5432
db_1              | 2021-04-02 20:12:49.033 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1              | 2021-04-02 20:12:49.043 UTC [84] LOG:  database system was shut down at 2021-04-02 20:12:48 UTC
db_1              | 2021-04-02 20:12:49.047 UTC [1] LOG:  database system is ready to accept connections
```
Any help would be greatly appreciated 🙏

**Comment by d4hines** [comment]
Gah! It was a CRLF issue. I re-checked out the repository with LF and made more progress, but not completely there:
```
Creating current-bench_db_1 ... done
Creating current-bench_pipeline_1       ... done
Creating current-bench_graphql-engine_1 ... done
Creating current-bench_db-migrate_1     ... done
Creating current-bench_frontend_1       ... done
Attaching to current-bench_db_1, current-bench_db-migrate_1, current-bench_pipeline_1, current-bench_graphql-engine_1, current-bench_frontend_1
db_1              | The files belonging to this database system will be owned by user "postgres".
db_1              | This user must also own the server process.
db_1              |
db_1              | The database cluster will be initialized with locale "en_US.utf8".
db_1              | The default database encoding has accordingly been set to "UTF8".
db_1              | The default text search configuration will be set to "english".
db_1              |
db_1              | Data page checksums are disabled.
db_1              |
db_1              | fixing permissions on existing directory /var/lib/postgresql/data ... ok
db_1              | creating subdirectories ... ok
db_1              | selecting dynamic shared memory implementation ... posix
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"server_configuration","info":{"live_query_options":{"batch_size":100,"refetch_delay":1},"transaction_isolation":"ISOLATION LEVEL READ COMMITTED","plan_cache_options":{"plan_cache_size":4000},"enabled_log_types":["http-log","websocket-log","startup","webhook-log","query-log"],"server_host":"HostAny","enable_allowlist":false,"log_level":"info","auth_hook_mode":null,"use_prepared_statements":true,"unauth_role":null,"stringify_numeric_types":false,"enabled_apis":["metadata","graphql","config","pgdump"],"enable_telemetry":true,"enable_console":true,"auth_hook":null,"jwt_secret":null,"cors_config":{"allowed_origins":"*","disabled":false,"ws_read_cookie":null},"console_assets_dir":null,"admin_secret_set":true,"port":8080}}}
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"postgres_connection","info":{"retries":1,"database_url":"postgres://docker:...@db:5432/docker"}}}
db_1              | selecting default max_connections ... 100
db_1              | selecting default shared_buffers ... 128MB
pipeline_1        |    application [INFO] Logging ready.
pipeline_1        |    application [ERROR] Database connection error:
pipeline_1        | Connection failure: could not connect to server: Connection refused
pipeline_1        |     Is the server running on host "db" (172.18.0.2) and accepting
pipeline_1        |     TCP/IP connections on port 5432?
pipeline_1        |
db_1              | selecting default time zone ... Etc/UTC
pipeline_1        | pipeline: internal error, uncaught exception:
pipeline_1        |           Postgresql.Error(_)
pipeline_1        |           Raised at Pipeline__Db_util.check_connection in file "lib/db_util.ml", line 10, characters 6-15
pipeline_1        |           Called from Pipeline.v in file "lib/pipeline.ml", line 245, characters 2-36
pipeline_1        |           Called from Cmdliner_term.app.(fun) in file "cmdliner_term.ml", line 25, characters 19-24
pipeline_1        |           Called from Cmdliner.Term.run in file "cmdliner.ml", line 117, characters 32-39
db_1              | creating configuration files ... ok
db_1              | running bootstrap script ... ok
db_1              | performing post-bootstrap initialization ... ok
db_1              | syncing data to disk ... ok
db_1              |
db_1              |
db_1              | Success. You can now start the database server using:
db_1              |
db_1              |     pg_ctl -D /var/lib/postgresql/data -l logfile start
db_1              |
db_1              | initdb: warning: enabling "trust" authentication for local connections
db_1              | You can change this by editing pg_hba.conf or using the option -A, or
db_1              | --auth-local and --auth-host, the next time you run initdb.
db_1              | waiting for server to start....2021-04-02 20:42:28.136 UTC [47] LOG:  starting PostgreSQL 12.6 (Debian 12.6-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
db_1              | 2021-04-02 20:42:28.138 UTC [47] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1              | 2021-04-02 20:42:28.162 UTC [48] LOG:  database system was shut down at 2021-04-02 20:42:27 UTC
db_1              | 2021-04-02 20:42:28.165 UTC [47] LOG:  database system is ready to accept connections
db_1              |  done
db_1              | server started
db_1              | CREATE DATABASE
db_1              |
db_1              |
db_1              | /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/init.sql
db_1              | GRANT
db_1              |
db_1              |
db_1              | 2021-04-02 20:42:28.440 UTC [47] LOG:  received fast shutdown request
db_1              | waiting for server to shut down....2021-04-02 20:42:28.441 UTC [47] LOG:  aborting any active transactions
db_1              | 2021-04-02 20:42:28.442 UTC [47] LOG:  background worker "logical replication launcher" (PID 54) exited with exit code 1
db_1              | 2021-04-02 20:42:28.442 UTC [49] LOG:  shutting down
db_1              | 2021-04-02 20:42:28.455 UTC [47] LOG:  database system is shut down
db_1              |  done
db_1              | server stopped
db_1              |
db_1              | PostgreSQL init process complete; ready for start up.
db_1              |
db_1              | 2021-04-02 20:42:28.549 UTC [1] LOG:  starting PostgreSQL 12.6 (Debian 12.6-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
db_1              | 2021-04-02 20:42:28.549 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
db_1              | 2021-04-02 20:42:28.549 UTC [1] LOG:  listening on IPv6 address "::", port 5432
db_1              | 2021-04-02 20:42:28.553 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1              | 2021-04-02 20:42:28.563 UTC [84] LOG:  database system was shut down at 2021-04-02 20:42:28 UTC
db_1              | 2021-04-02 20:42:28.567 UTC [1] LOG:  database system is ready to accept connections
db-migrate_1      | omigrate: [INFO] Database already exists
db-migrate_1      | omigrate: [INFO] Creating the migrations table
db-migrate_1      | omigrate: [INFO] Applying up migration 20210013150054
db-migrate_1      | omigrate: [INFO] Applying up migration 20210101173805
db-migrate_1      | omigrate: [INFO] Applying up migration 20210202135643
current-bench_db-migrate_1 exited with code 0
graphql-engine_1  | {"type":"pg-client","timestamp":"2021-04-02T20:42:28.484+0000","level":"warn","detail":{"message":"postgres connection failed, retrying(0)."}}
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"catalog_migrate","info":"Successfully initialized the catalog (at version 40)."}}
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"schema-sync","info":{"thread_id":"ThreadId 40","instance_id":"f782e680-b498-41ed-9fb8-18278193f751","message":"listener thread started"}}}        
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"schema-sync","info":{"thread_id":"ThreadId 41","instance_id":"f782e680-b498-41ed-9fb8-18278193f751","message":"processor thread started"}}}       
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"event_triggers","info":"preparing data"}}
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"event_triggers","info":"starting workers"}}
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"scheduled_triggers","info":"preparing data"}}
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"telemetry","info":"Help us improve Hasura! The graphql-engine server collects anonymized usage stats which allows us to keep improving Hasura at warp speed. To read more or opt-out, visit https://hasura.io/docs/1.0/graphql/manual/guides/telemetry.html"}}
graphql-engine_1  | {"type":"startup","timestamp":"2021-04-02T20:42:28.484+0000","level":"info","detail":{"kind":"server","info":{"time_taken":0.7339076,"message":"starting API server"}}}
```

Apparently `pipeline_1` fails because it can't connect to `db`.

**Comment by d4hines** [comment]
Just in case it was still a Windows-specific error, I tried from a natively-installed Linux distro, and got the same results.

**Comment by gs0510** [comment]
Hi @d4hines! Thanks for opening this issue, I'm looking into why the pipeline can't connect to the db and will update as soon as I have a solution or more insight. :)

**Comment by rizo** [comment]
The error message from the db isn't very descriptive unfortunately, maybe let's improve that first.

I find it strange that the migrations container did manage to connect to the db – they're using the same connection information.

**Comment by gs0510** [comment]
@rizo  I think the problem is that the pipeline container tries to connect to the database before the postgres server is setup. If you look at the logs, the pipeline errors out first and then the logs say that the database is created.  Maybe we should introduce a time lag before the pipeline tries to connect to the db?

Also, the pipeline does log the error [here](https://github.com/ocurrent/current-bench/blob/main/pipeline/lib/db_util.ml#L8).

**Comment by d4hines** [comment]
Indeed, adding `let () = Unix.sleep 5` at the top of `./pipeline/bin/main.ml` seems to have temporarily fixed the issue.

**Comment by gs0510** [comment]
Fixed by #91
